docs/test-anatomy-sample.md

```diff
@@ -0,0 +1,244 @@
+# Anatomy of a Sample Test
+
+In this example, we will analyze an already existing test on Go for checking
+logging [lim](../tests/lim/). This test returns success when receiving 1 message
+from the log.
+
+## Test Structure
+
+The test directory contains the expected files:
+
+* `Makefile`
+* `eden-config.yml` - eden config file
+* `eden.lim.tests.txt` - test scenario
+* source files for the test binary, in this case, `lim_test.go`
+* subscripts in the `testdata/` directory
+
+## Test Binary
+
+The test binary is written in go, as are most of the integration test.
+
+### Imports
+
+The test relies on processing data structures of which `eden` is aware, and so
+imports the relevant libraries
+[in the go file](../blob/1ef2175ccf68b01f02ec88f37adc662da2896be1/tests/lim/lim_test.go#L13-L16):
+
+```go
+import (
+...
+"github.com/lf-edge/eden/pkg/controller/eflowlog"
+"github.com/lf-edge/eden/pkg/controller/einfo"
+"github.com/lf-edge/eden/pkg/controller/elog"
+"github.com/lf-edge/eden/pkg/controller/emetric"
+...
+)
+```
+
+### Main Function
+
+As the binary is a go test file, with setup and teardown, the main function is
+[TestMain](../blob/1ef2175ccf68b01f02ec88f37adc662da2896be1/tests/lim/lim_test.go#L83-L142):
+
+```go
+func TestMain(m *testing.M) {
+    // lots of extra code
+}
+```
+
+You can read more about `TestMain` in the
+[go testing reference documentation](https://pkg.go.dev/testing#hdr-Main), and
+usage for EVE
+[on the EVE wiki](https://wiki.lfedge.org/display/EVE/EVE+Integration+Testing).
+
+### Test Functions
+
+In order to actually execute tests, you need to create test functions. These
+all have the signature `func Test*` (other than `TestMain`, of course).
+The lim testing file has four such functions:
+
+* [TestLog](../blob/1ef2175ccf68b01f02ec88f37adc662da2896be1/tests/lim/lim_test.go#L144)
```

Comment:
I think it is not a good idea to use such links. Can we somehow get rid of them?

Reply:
I thought about it, @giggsoff , but I wanted people to be able to see it easily, so I linked to a specific commit, rather than the main `master` branch. This way the link is permanent. Even when the code changes, the link will be good.